### PR TITLE
Fix propagateEnv = false to not propagate env

### PR DESCRIPTION
**Pull request by lolgab**
Originated from https://github.com/com-lihaoyi/mill/issues/2881

Pull request: https://github.com/com-lihaoyi/os-lib/pull/238

### CODE REVIEW

File: os/src-jvm/ProcessOps.scala [hunk 1]
```diff
@@ -337,12 +337,17 @@ private[os] object ProcessOps {
...
-    val baseEnv =
-      if (propagateEnv) sys.env
-      else Map()
-    for ((k, v) <- baseEnv ++ Option(env).getOrElse(Map())) {
-      if (v != null) builder.environment().put(k, v)
-      else builder.environment().remove(k)
+    val environment = builder.environment()
+
+    if (!propagateEnv) {
+      environment.clear()
```
Comment (lolgab) [inline os/src-jvm/ProcessOps.scala:343]: `java.lang.ProcessBuilder` propagates the environment by default. If you don't want to propagate it you need to `clear()` the map.

File: os/test/src-jvm/SubprocessTests.scala [hunk 2]
```diff
@@ -104,17 +104,41 @@ object SubprocessTests extends TestSuite {
...
-        assert(res1.out.lines() == Seq("Hello12"))
+        locally {
+          val res3 = proc("bash", "-c", "echo 'Hello'$ENV_ARG").call(env = Map("ENV_ARG" -> "123"))
+          assert(res3.out.lines() == Seq("Hello123"))
+        }
 
-        val res2 = proc("bash", "-c", "echo 'Hello$ENV_ARG'").call(env = Map("ENV_ARG" -> "12"))
-        assert(res2.out.lines() == Seq("Hello$ENV_ARG"))
+        locally {
+          val res4 = proc("bash", "-c", "echo \"$TEST_SUBPROCESS_ENV\"").call(
```
Comment (lefou) [inline os/test/src-jvm/SubprocessTests.scala:128]: Can you add a comment, that we expect this env var to be set from the build tool / test runner?

**Review by lefou — COMMENTED** [review]
Nice catch! Looks good to me. One nit below.

File: os/src-jvm/ProcessOps.scala (hunk 1 above)
Comment (lefou) [reply os/src-jvm/ProcessOps.scala:343]: Or even assert that it is set as expected.

File: os/test/src-jvm/SubprocessTests.scala (hunk 2 above)
Comment (lefou) [reply os/test/src-jvm/SubprocessTests.scala:128]: Or even assert that it is set as expected.

**Review by lefou — APPROVED** [review]

**Pull request merged**